test/anttasks/org/apache/cassandra/anttasks/TestNameCheckTask.java

```diff
@@ -158,14 +161,21 @@ private Class<?> normalize(Class<?> klass)
      */
     private Stream<? extends Class<?>> expand(Class<?> klass, Reflections reflections)
     {
-        Set<? extends Class<?>> subTypes = reflections.getSubTypesOf(klass);
-        if (subTypes == null || subTypes.isEmpty())
-            return Stream.of(klass);
-        Stream<? extends Class<?>> subs = subTypes.stream();
-        // assume we include if not abstract
-        if (!Modifier.isAbstract(klass.getModifiers()))
-            subs = Stream.concat(Stream.of(klass), subs);
-        return subs;
+        Set<Class<?>> concreteTypes = new HashSet<>();
+        Deque<Class<?>> typeStack = new ArrayDeque<>();
+        typeStack.push(klass);
+
+        while (!typeStack.isEmpty())
+        {
+            Class<?> type = typeStack.pop();
+
+            if (!Modifier.isAbstract(type.getModifiers()))
+                concreteTypes.add(type);
+
+            reflections.getSubTypesOf(type).forEach(typeStack::push);
+        }
+
+        return concreteTypes.stream();
```

Comment:
@jacek-lewandowski @smiklosovic This was causing issues for `ValueThresholdTester`, which is a subclass of `ThresholdTester` but is also `abstract`. Basically we can handle more than one level of inheritance now. WDYT?

Reply:
CC @bereng

Reply:
Sgtm, you're just walking the whole tree.

Reply:
Bonus: Getting to use a `Deque` for something :D